### ISSUE TITLE
Download data exports as Excel / CSV

### ISSUE THREAD

**Issue by leeensminger** [leeensminger/DelDOT-NPDES-Viewer#1]

**Comment by leeensminger** [comment]
@dobrochtaKCI Please create 2 to 5 mockups in Balsamiq detailing how this new workflow might work and let's review offline.

**Comment by dobrochtaKCI** [comment]
[Export_CSV.pdf](https://github.com/leeensminger/DelDOT-NPDES-Viewer/files/1904793/Export_CSV.pdf)

**Comment by kanglu666** [comment]
in progress...
1: nee to update the UI work flow
2: build new service to down load data as CSV

**Comment by kanglu666** [comment]
Download data exports as Excel / CSV  function has been added into web viewer, ready for testing.

**Comment by kanglu666** [comment]
open excel, load the .csv file from  data-->From Text/CSV.   everything will be fine.

the reason i don't use comma is because there are comment and shape fields in the data, the shape value will export as coordinates string :
SRID=2235;POLYGON ((599657.3948174119 242448.38616774231, 599620.364379663 242441.44294015318, 599565.74276974425 242422.64081240445, 599508.22090316191 242396.33016949147, 599450.18361766264 242365.07957582176, 599462.31220233068 242344.30599532276, 599514.20317466184 242372.20062457025, 599596.20268257707 242403.81243798882, 599652.36825266108 242423.58273965865, 599657.3948174119 242448.38616774231))

you will see this string contains "," and ";",  if using comma, the excel will not work right.



So i will recommend just using the tab.

**Comment by dobrochtaKCI** [comment]
I agree this makes sense based on the fields you described. The tool appears to be working for all of the queries, I have tested except when the features returned is significantly large (thousands of records returned). Is there a known limit to the number of records that can be returned successfully? I have been successful returning up to about 3000 records or so, but anything larger than that returns a blank export.

**Comment by leeensminger** [comment]
@kanglu666 see question above

**Comment by kanglu666** [comment]
after testing, it is the sql server connection timeout issue. i will make the code updates and bump the time out limit.

**Comment by kanglu666** [comment]
updated, ready for testing in dev.

updated SQL query, and increase the sql command timeout number.

**Comment by leeensminger** [comment]
Looks ok in dev.  Exported 88,000+ records to csv and all records are shown in the csv.

**Comment by cstevenson1** [comment]
Typically when you export to excel, it names the file Structures, Conveyances or BMPS. However there are a number of scenarios that will generate files titled Export. These files do not open/work.

Examples (there are more beyond these examples):  
1. Using Geographic Filter to query BMPs in New Castle
2. Using Advanced Query Builder to query Swale Connections in the Army Creek Watershed
3. Using Advanced Query Builder to query Pipe Ends with a Owner of Dover

I did not test every combination but it does not seem seem field specific, i.e. all queries in advanced query builder for pipe ends don't work, not just if you search by Owner. It also doesn't seem specific to joined tables because it appears to be working if you query the Inlets layer. Please let me know if these observations are not consistent with your findings and I need to adjust my workflow.

**Comment by leeensminger** [comment]
@kanglu666 looks like this is correct, the queries above result in an "export" csv that is empty. Thoughts?

**Comment by kanglu666** [comment]
1: Using Geographic Filter to query BMPs in New Castle
This one has been fixed, updated in PROD.

will test and work on other query combinations

**Comment by kanglu666** [comment]
updated, ready for testing in both DEV and PROD.

NOTE: i would suggest that someone test all query combinations in dev to make sure we are not missing anything,

**Comment by leeensminger** [comment]
These three work in prod.

@kanglu666 please fix Control Structures where Watershed = Army Creek

**Comment by leeensminger** [comment]
Brendan, this was basically done, but needs some minor bug squashing

**Comment by cstevenson1** [comment]
KCI Newark will continue to monitor the errors and report back to GSG
